sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -996,13 +1001,15 @@ async Task performProcessing()
 #pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
                     }
                 }
-                catch (Exception ex) when ((ex is OperationCanceledException) || (ex is TaskCanceledException))
+                catch (TaskCanceledException)
```

Comment:
In theory, we shouldn't see an `OperationCanceledException` instance here because we wrap them into a `TaskCanceledException` for known cancellations, so the only ones that should get to this level are most likely an indication that something went wrong at the service communication level within the AMQP library that its built on.   

For a bit wider context, the [service documentation](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions) calls out that operation cancelled indicates that somewhere you closed or disposed something.  Because of this, we're trying to avoid confusion when normal and expected cancellation scenarios happen by consistently surfacing only `TaskCanceledException` rather than being agnostic to both.